src/controllers/productsController.ts

```diff
@@ -12,107 +12,125 @@ dotenv.config();
 
 applyFileSysyem();
 
+
 export const getAllProducts = async (req: Request, res: Response) => {
-  let page = Number(req.query.page) || 1;
-  let limit = Number(req.query.limit) || 9;
+  try {
+    let page = Number(req.query.page) || 1;
+    let limit = Number(req.query.limit) || 9;
 
-  if (page < 0) {
-    page = 1;
-  }
-  if (limit > 80 || limit < 0) {
-    limit = 9;
-  }
+    if (page < 0) {
+      page = 1;
+    }
+    if (limit > 80 || limit < 0) {
+      limit = 9;
+    }
 
-  const search = req.query.search?.toString();
-  let conditions: any = {};
+    const search = req.query.search?.toString();
+    let conditions: any = {};
 
-  if (search) {
-    conditions.name = {
-      [Op.like]: `%${req.query.search}%`,
-    };
-  }
+    if (search) {
+      conditions.name = {
+        [Op.like]: `%${req.query.search}%`,
+      };
+    }
 
-  let greaterThan;
-  if (!Number.isNaN(req.query["gte"])) {
-    greaterThan = Number(req.query["gte"]);
-  }
+    let greaterThan;
+    if (!Number.isNaN(req.query['gte'])) {
+      greaterThan = Number(req.query['gte']);
+    }
 
-  let lessThan;
-  if (!Number.isNaN(req.query["lte"])) {
-    lessThan = Number(req.query["lte"]);
-  }
+    let lessThan;
+    if (!Number.isNaN(req.query['lte'])) {
+      lessThan = Number(req.query['lte']);
+    }
 
-  if (greaterThan) {
-    conditions = {
-      ...conditions,
-      price: {
-        [Op.gte]: greaterThan,
-      },
-    };
-  }
+    if (greaterThan) {
+      conditions = {
+        ...conditions,
+        price: {
+          [Op.gte]: greaterThan,
+        },
+      };
+    }
 
-  if (lessThan) {
-    conditions = {
-      ...conditions,
-      price: {
-        ...conditions["price"],
-        [Op.lte]: lessThan,
-      },
-    };
-  }
+    if (lessThan) {
+      conditions = {
+        ...conditions,
+        price: {
+          ...conditions['price'],
```

Comment:
What is the purpose of this ? and why it is not used in the greaterThan block ?

Reply:
this one also was done by Mohammad

Reply:
The purpose is in case the user wants to choose greatThan and lessThan at same time to have them both, so he select between (10 and 200) and etc , if he wants (price > 100) he also can do it, if he wants (price < 100) he also can do it, with this implementation.